extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeImage.php

```diff
@@ -12,41 +12,118 @@ class NodeImage extends Node {
 
 	public function getData() {
 		if ( !isset( $this->data ) ) {
-			$imageData = $this->getRawValueWithDefault( $this->xmlNode );
-
-			if( is_string($imageData) && PortableInfoboxDataBag::getInstance()->getGallery($imageData)) {
-				$imageData = PortableInfoboxDataBag::getInstance()->getGallery($imageData);
+			$this->data = array();
+
+			$parameterValue = $this->getRawValueWithDefault( $this->xmlNode );
+
+			if ( $this->containsTabberOrGallery( $parameterValue ) ) {
+				$parsed = $this->getExternalParser()->parseRecursive( $parameterValue );
+
+				$items = $this->getGalleryItems( $parsed ) + $this->getTabberItems( $parsed );
+				for( $i = 0; $i < count( $items ); $i++ ) {
+					$this->data[] = $this->getImage(
+						$items[$i]['title'],
+						$items[$i]['caption'],
+						$items[$i]['caption']
+					);
+				}
+			} else {
+				$this->data[] = $this->getImage(
+					$parameterValue,
+					$this->getValueWithDefault( $this->xmlNode->{self::ALT_TAG_NAME} ),
+					$this->getValueWithDefault( $this->xmlNode->{self::CAPTION_TAG_NAME} )
+				);
 			}
+		}
+
+		return $this->data;
+	}
+
+	private function getGalleryItems( $html ) {
+		$items = array();
+
+		if ( preg_match( '#\sdata-model="([^"]+)"#', $html, $galleryOut ) ) {
+			$model = json_decode( htmlspecialchars_decode( $galleryOut[1] ), true );
+			$items = array_map( function( $modelItem ) {
+				return array(
+					'title' => $modelItem['title'],
+					'caption' => $modelItem[ 'caption' ]
+				);
+			}, $model );
+		}
 
-			$title = $this->getImageAsTitleObject( $imageData );
-			$file = $this->getFilefromTitle( $title );
-			if ( $title instanceof \Title ) {
-				$this->getExternalParser()->addImage( $title->getDBkey() );
+		if ( preg_match_all('#data-image-key="([^"]+)".*?\s<h2>(.*?)<\/h2>#is', $html, $galleryOut ) ) {
+			for( $i = 0; $i < count( $galleryOut[0] ); $i++ ) {
+				$items[] = array(
+					'title' => $galleryOut[1][$i],
+					'caption' => $galleryOut[2][$i]
+				);
 			}
-			$ref = null;
-			$alt = $this->getValueWithDefault( $this->xmlNode->{self::ALT_TAG_NAME} );
-			$caption = $this->getValueWithDefault( $this->xmlNode->{self::CAPTION_TAG_NAME} );
-
-			$this->data = [
-				'url' => $this->resolveImageUrl( $file ),
-				'name' => ( $title ) ? $title->getText() : '',
-				'key' => ( $title ) ? $title->getDBKey() : '',
-				'alt' => $alt,
-				'caption' => $caption
-			];
-
-			if ( $this->isVideo( $file ) ) {
-				$this->data = $this->videoDataDecorator( $this->data, $file );
+		}
+
+		return $items;
+	}
+
+	private function getTabberItems( $html ) {
```

Comment:
Handling that differently would need more in-depth work inside of the parsing flow, similar to what the ArticleAsJson does. For now this should work, though it needs some tests both for extraction (unit) and to notify if expected format is returned by parser from tabber (one integration test)